character.cpp

```diff
@@ -0,0 +1,39 @@
+#include "character.h"
+
+Character::Character(std::string name, int health, unsigned int damage) :
+                    name(name),
+                    health(health),
+                    damage(damage) {}
+
+std::string Character::GetName() const{
+  return name;
+}
+
+int Character::GetHealth() const{
+  return health;
+}
+
+unsigned int Character::GetDamage() const{
+  return damage;
+}
+
+bool Character::IsAlive(){
+  return health > 0;
+}
+
+void Character::TakeDamage(const int damage){
+  health -= damage;
+  if(health < 0){
+    health = 0;
+  }
+}
+
+void Character::Attack(std::shared_ptr<Character> enemy){
```

Comment:
miert hasznalsz shared_ptr-t? also: const?

Reply:
nem lehet használni?

Reply:
De. Lehet hasznalni a std::map-et is. meg lambda fuggvenyeket. constexpr-et, stb... :-) Ami C++ azt lehet. De nem is azt irtam, hogy nem szabad hasznalni, hanem megkerdeztem, hogy miert hasznalod itt?

Reply:
ja félreértelmeztem, bocsi. utólag belegondolva tényleg nem is kell.

Reply:
:smile: